File: samples/vbnet/connector_net_intro/connector_net_intro/Form1.vb
Imports MySql.Data.MySqlClient
Public Class Form1
  Dim myconn As MySqlConnection

  Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
    Try
      myconn = New MySqlConnection( _
          "Data Source=192.168.80.128;Initial Catalog=mylibrary;" + _
          "User ID=root;PWD=uranus")
      myconn.Open()
    Catch myerror As MySqlException
      MsgBox("Database connection error: " & myerror.Message)
      Me.Close()
    End Try
    Button2.Enabled = True
  End Sub

  Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
    Dim com As MySqlCommand
    Dim dr As MySqlDataReader
    com = New MySqlCommand( _
      "SELECT publID, publName FROM publishers ORDER BY publName", _
      myconn)
    dr = com.ExecuteReader()
    While dr.Read()
      TextBox1.AppendText("id = " & dr!publID & ",  name = " & dr!publName & vbCrLf)
    End While
    dr.Close()
  End Sub


  Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
    Me.Close()
  End Sub
End Class

File: samples/csharp/connector_net_intro/connector_net_intro/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

#endregion

namespace connector_net_intro {
  partial class Form1 : Form {
    MySqlConnection myconn;
    public Form1() {
      InitializeComponent();
    }

    private void button1_Click(object sender, EventArgs e) {
      try {
        myconn = new MySqlConnection(
          "Data Source=192.168.80.128;Initial Catalog=mylibrary;" + 
          "User ID=root;PWD=uranus");
        myconn.Open();
      }
      catch (MySqlException myerror) {
        MessageBox.Show("MySQL connection error: " + myerror.Message);
        this.Close();
      }
      button2.Enabled = true;
    }

    private void button2_Click(object sender, EventArgs e) {
      MySqlCommand com;
      MySqlDataReader dr;
      com = new MySqlCommand( 
        "SELECT publID, publName FROM publishers ORDER BY publName", 
        myconn);
      dr = com.ExecuteReader();
      while (dr.Read())
        textBox1.AppendText("id = " + dr["publID"] +
          ",  name = " + dr["publName"] + Environment.NewLine);
      dr.Close();
    }

    private void button3_Click(object sender, EventArgs e) {
      this.Close();
    }
  }
}